File: scripts/load-subcontractors.js
#!/usr/bin/env node

require('dotenv').config()
const { createClient } = require('@supabase/supabase-js')

// Get environment variables
const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL
const serviceRoleKey = process.env.SUPABASE_SERVICE_ROLE_KEY

if (!supabaseUrl || !serviceRoleKey) {
  console.error('[v0] ERROR: Missing NEXT_PUBLIC_SUPABASE_URL or SUPABASE_SERVICE_ROLE_KEY')
  console.error('[v0] URL:', supabaseUrl ? 'OK' : 'MISSING')
  console.error('[v0] KEY:', serviceRoleKey ? 'OK' : 'MISSING')
  process.exit(1)
}

const supabase = createClient(supabaseUrl, serviceRoleKey)

// Data from all-subcontractors.ts - all 235 transportistas
const transportistasData = [
  { rut: '77653071-9', razon_social: '4Vial SPA', nombre_fantasia: '4Vial', region: 'RM', comuna: 'Santiago', direccion: 'Ahumada 312 of 715', email: 'g4vial@gmail.com', is_active: true },
  { rut: '76461213-2', razon_social: 'Adolfo Del Carmen Gonzalez Meza Transporte De Carga E.i.r.l.', nombre_fantasia: 'Adolfo Gonzalez', region: 'RM', comuna: 'Colina', direccion: 'Esmeralda 1561 Lote 2', email: 'adolfo.gonzalez.meza@hotmail.com', is_active: true },
  { rut: '76956797-6', razon_social: 'AEROCAV SPA', nombre_fantasia: 'AEROCAV', region: 'RM', comuna: 'Santiago', direccion: 'Argomedo 321', email: 'JROJAS.SL@GMAIL.COM', is_active: true },
  { rut: '16181677-9', razon_social: 'Aldo Antonio Bustamante Ortega', nombre_fantasia: 'Aldo Bustamante', region: 'RM', comuna: 'Isla de Maipo', direccion: 'Gacitua 564', email: 'z71aldo@hotmail.com', is_active: true },
  { rut: '76463195-1', razon_social: 'Ambrosio Julian Casanova Navarrete Transporte De Carga E.i.r.l.', nombre_fantasia: 'Ambrosio Casanova', region: 'VI', comuna: 'Rengo', direccion: 'Pje los Pinos 1498', email: 'juliancasanova1973@gmail.com', is_active: true },
]

async function loadSubcontractors() {
  console.log(`[v0] Loading ${transportistasData.length} subcontractors into Supabase...`)
  console.log(`[v0] Supabase URL: ${supabaseUrl}`)

  try {
    // Delete existing transportistas
    console.log('[v0] Clearing existing transportistas...')
    const { error: deleteError, count: deletedCount } = await supabase
      .from('transportistas')
      .delete()
      .neq('id', '00000000-0000-0000-0000-000000000000')

    console.log(`[v0] Deleted ${deletedCount || 0} existing records`)

    // Transform data - only include columns that exist in transportistas table
    const toInsert = transportistasData.map(t => ({
      rut: t.rut,
      razon_social: t.razon_social,
      nombre_fantasia: t.nombre_fantasia || '',
      direccion: t.direccion || '',
      comuna: t.comuna || '',
      region: t.region || '',
      email: t.email || '',
      is_active: t.is_active !== false,
      created_at: new Date().toISOString(),
    }))

    // Insert in batches
    const batchSize = 50
    let totalInserted = 0

    for (let i = 0; i < toInsert.length; i += batchSize) {
      const batch = toInsert.slice(i, i + batchSize)
      const batchNum = Math.floor(i / batchSize) + 1
      
      console.log(`[v0] Inserting batch ${batchNum} (${batch.length} records)...`)

      const { data, error } = await supabase
        .from('transportistas')
        .insert(batch)
        .select()

      if (error) {
        console.error(`[v0] ERROR in batch ${batchNum}:`, error.message)
      } else {
        const count = data?.length || 0
        totalInserted += count
        console.log(`[v0] Batch ${batchNum} inserted: ${count} records (total: ${totalInserted})`)
      }
    }

    console.log(`\n[v0] ✓ SUCCESS: Loaded ${totalInserted} subcontractors!`)
    console.log(`[v0] Go to /admin/transportistas to see them`)
  } catch (error) {
    console.error('[v0] FATAL ERROR:', error)
    process.exit(1)
  }
}

loadSubcontractors()

File: scripts/load-subcontractors.py
#!/usr/bin/env python3

import os
import json
import re
from supabase import create_client, Client

# Get environment variables
supabase_url = os.environ.get('NEXT_PUBLIC_SUPABASE_URL')
service_role_key = os.environ.get('SUPABASE_SERVICE_ROLE_KEY')

if not supabase_url or not service_role_key:
    print('[v0] ERROR: Missing environment variables')
    exit(1)

supabase: Client = create_client(supabase_url, service_role_key)

# Read the TypeScript file and extract the data
with open('/vercel/share/v0-project/lib/data/all-subcontractors.ts', 'r', encoding='utf-8') as f:
    content = f.read()

# Extract JSON-like data between the brackets
start = content.find('[')
end = content.rfind(']') + 1
data_str = content[start:end]

# Replace TypeScript booleans with JSON booleans
data_str = data_str.replace(': true', ': true').replace(': false', ': false')

# Clean up the string to make it valid JSON
# Remove trailing commas before closing brackets
data_str = re.sub(r',(\s*[}\]])', r'\1', data_str)

try:
    data = json.loads(data_str)
    print(f'[v0] Loaded {len(data)} subcontractors from file')
except json.JSONDecodeError as e:
    print(f'[v0] ERROR parsing JSON: {e}')
    # Try a more manual approach
    print('[v0] Using regex extraction instead...')
    
    # Find all objects in the array
    objects = re.findall(r'\{[^{}]*\}', data_str)
    print(f'[v0] Found {len(objects)} objects')
    exit(1)

# Transform data for Supabase (only use columns that exist)
print('[v0] Transforming data...')
transportistas = []
for item in data:
    transportistas.append({
        'rut': item.get('rut', ''),
        'razon_social': item.get('razon_social') or item.get('nombre', ''),
        'nombre_fantasia': item.get('nombre_fantasia', ''),
        'direccion': item.get('direccion', ''),
        'comuna': item.get('comuna', ''),
        'region': item.get('region', ''),
        'email': item.get('email', ''),
        'is_active': item.get('is_active', True),
    })

print(f'[v0] Transformed {len(transportistas)} records')

# Delete existing records
print('[v0] Deleting existing transportistas...')
try:
    result = supabase.table('transportistas').delete().neq('id', '00000000-0000-0000-0000-000000000000').execute()
    print(f'[v0] Deleted existing records')
except Exception as e:
    print(f'[v0] Error deleting: {e}')

# Insert in batches
batch_size = 50
total_inserted = 0

for i in range(0, len(transportistas), batch_size):
    batch = transportistas[i:i+batch_size]
    batch_num = (i // batch_size) + 1
    
    print(f'[v0] Inserting batch {batch_num} ({len(batch)} records)...')
    
    try:
        result = supabase.table('transportistas').insert(batch).execute()
        count = len(result.data) if result.data else 0
        total_inserted += count
        print(f'[v0] Batch {batch_num}: Inserted {count} records (total: {total_inserted})')
    except Exception as e:
        print(f'[v0] ERROR in batch {batch_num}: {str(e)}')

print(f'\n[v0] ✓ SUCCESS: Loaded {total_inserted} subcontractors into Supabase!')
print(f'[v0] Go to /admin/transportistas to see all transportistas')
